layouts/partials/blocks/awards.html

```diff
@@ -0,0 +1,269 @@
+              console.warn("No year list or awards container found.");
+              return;
+            }
+
+            const yearItems = yearList.querySelectorAll("li");
+            yearItems.forEach((item) => {
+              item.addEventListener("click", () => {
+                const activeYear = item.textContent.trim();
+
+                yearItems.forEach((li) => li.classList.remove("active"));
+                item.classList.add("active");
+
+                const awardSections =
+                  awardsContainer.querySelectorAll(".award-entity");
+                awardSections.forEach((award) => {
+                  const awardYear = award.getAttribute("data-year");
+                  award.style.display =
+                    awardYear === activeYear ? "block" : "none";
+                });
+
+                const sections = [
+                  {
+                    header: ".national-awards-header",
+                    container: ".national-awards",
+                  },
+                  {
+                    header: ".national-finalists-header",
+                    container: ".national-finalists",
+                  },
+                  {
+                    header: ".state-awards-header",
+                    container: ".state-awards",
+                  },
+                  {
+                    header: ".state-finalists-header",
+                    container: ".state-finalists",
+                  },
+                ];
+
+                sections.forEach(({ header, container }) => {
+                  const section = awardsContainer.querySelector(container);
+                  const headerElement = awardsContainer.querySelector(header);
+                  const hasVisibleAwards = Array.from(
+                    section.querySelectorAll(".award-entity"),
+                  ).some((award) => award.style.display === "block");
+                  headerElement.style.display = hasVisibleAwards
+                    ? "block"
+                    : "none";
+                });
+              });
+            });
+
+            yearItems[0]?.click();
+          });
+        </script>
+
+        <div
+          id="awards-padding"
+          style="padding: 15px 20px 15px 25px; display: flex; flex-wrap: wrap; width: 100%;"
+        >
+          <div
+            class="awards-container"
+            style="flex: 1 1 100%; word-break: break-word;"
+          >
+            <p class="award-national-state national-awards-header">
+              National Awards
+            </p>
+            <div class="national-awards">
+              {{ range where (where . "national_or_state" "National") "winner_or_finalist_honorable_mention" "Winner" }}
+                <div class="award-entity" data-year="{{ .year }}">
+                  {{ .entity }}:
+                  <span class="award-category">{{ .award_category }}</span>
+                  <p class="award-story">
+                    {{ if .link }}
```

Comment:
We have a helper partial because this comes up a lot: 

```
{{ partial "tag/maybe-link.html" (dict
      "link" .link
      "text" .story
      "class" "award-story-link"
  )
}}
```

Reply:
Oh, this should replace the `{{ if .link }}` and `{{ else }}` code. It's just a shortcut for the "if .link use an `a`, else just dump the text" pattern.